docs/nmt-lib.md

```diff
@@ -0,0 +1,187 @@
+# Namespace Merkle Tree Library
+The Namespace Merkle Tree (NMT) library implements the NMT data structure outlined in the [NMT spec](./spec/nmt.md). 
+In the following sections, we will provide instructions on how to utilize the library to construct an NMT and offer insights into its fundamental methods.
+
+## NMT Initialization and Configuration
+
+An NMT can be constructed using the `New` function.
+```go
+func New(h hash.Hash, setters ...Option) *NamespacedMerkleTree
+```
+It receives a base hash function alongside with some optional configurations, namely:
+1. Namespace ID byte-size: If not specified then a default maximum is applied by the library.
+2. The initial capacity of the tree i.e., the number of leaves: if not specified, a default maximum is applied.
```

Comment:
[question] the default capacity doesn't seem like a maximum b/c the tree can grow past the initial capacity

```suggestion
2. The initial capacity of the tree i.e., the number of leaves: if not specified, a default is applied.
```

Reply:
You're correct that the tree can exceed its initial capacity. It's worth noting that any necessary checks should have been performed in the [Push](https://github.com/celestiaorg/nmt/blob/3bdcd9ca7c10600b6b8db21a9a3e1984770aa218/nmt.go#L397) method, but this is not the case.

Reply:
Addressed https://github.com/celestiaorg/nmt/pull/120/commits/9d50cafbc6dea994a29bc7a768e23d7c35106761